pkg/kv/kvserver/app_batch.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package kvserver
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverpb"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/raftlog"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
+)
+
+type appBatch struct {
+	// TODO(tbg): this will absorb the following fields from replicaAppBatch:
+	//
+	// - batch
+	// - state
+	// - changeRemovesReplica
+}
+
+func (b *appBatch) assertAndCheckCommand(
```

Comment:
Method comments would help. Maybe `appBatch` comment/overview too. I.e. why all these stages? What's the input/output of each stage (or whole sequence of stages) semantically?

Reply:
https://github.com/cockroachdb/cockroach/pull/93266 describes the goal:

----

This long (but individually small) sequence of commits moves (*replicaAppBatch).Stage close to the structure that was prototyped in https://github.com/cockroachdb/cockroach/pull/93265, where it has the following steps:

command checks (standalone)
testing interceptors (replica)
pre-add triggers (standalone)
pre-add triggers (replica)
add (to pebble batch, standalone)
post-add triggers (standalone)
post-add triggers (replica)

----

I adopted this comment and put it on `appBatch`. I am planning on expanding the comments significantly in the PR that establishes a functional standalone `appBatch` but let me know pieces that you're confused about today and I'll make sure they get some love earlier.